nvflare/fuel/f3/stream_cell.py

```diff
@@ -0,0 +1,209 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+from typing import Callable
+
+from nvflare.fuel.f3.cellnet.cell import Cell
+from nvflare.fuel.f3.message import Message
+from nvflare.fuel.f3.streaming.blob_streamer import BlobStreamer
+from nvflare.fuel.f3.streaming.byte_receiver import ByteReceiver
+from nvflare.fuel.f3.streaming.byte_streamer import ByteStreamer
+from nvflare.fuel.f3.streaming.file_streamer import FileStreamer
+from nvflare.fuel.f3.streaming.object_streamer import ObjectStreamer
+from nvflare.fuel.f3.streaming.stream_types import ObjectIterator, ObjectStreamFuture, Stream, StreamError, StreamFuture
+
+
+class StreamCell:
```

Comment:
The StreamCell class holds 4 different kinds of Streamer, each performs separate independent type of functions, ByteStream, BlobStream, FileStream and ObjectStream. Every time when creating a StreamCell instance, it will create other unrelated streamer. Probably it's better to separate the different types of StreamCell, or use the factory pattern to create different type of streamer as needed.

Also is it possible to abstract the Streamer class, and make the ByteStreamer, BlobStreamer, FileStreamer and ObjectStreamer extend from it? Conceptually I think that's supposed to be. But right now they are all separately without any relationship.